lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1398,7 +1398,10 @@ def cmd_sessions(*args)
           unless script
             method = 'script'
             script = val
+            # Assume any parameter after the script name is a flag/argument we want to pass to the script itself.
+            args.excluding(val, opt).each { |a| extra << a }
```

Comment:
This seems like a bad approach. Let me show you why I think this is the case.

Modification to code to make it easier to debug:

```
        # Run a script or module on specified sessions
        when "-s", "--script"
          unless script
            p args
            p val
            p opt
            method = 'script'
            script = val
          end
        # Upload and exec to the specific command session
```

And here is the output from testing:

```
msf6 payload(windows/x64/meterpreter/reverse_tcp) > sessions -i 1 -s scripts/meterpreter/winenum.rb
["-i", "1", "-s", "scripts/meterpreter/winenum.rb"]
"scripts/meterpreter/winenum.rb"
"-s"
[-] Invalid session identifier: 1
msf6 payload(windows/x64/meterpreter/reverse_tcp) > 
```

So basically your passing in every argument other than the `-s` and the script name into the script itself. Rather than passing in any parameter after the script name itself, you are instead passing in ALL parameters other than the script name itself and the `-s` argument. It just so happens that the script doesn't have a `-i` option so it ends up ignoring that in your case, but this might not be true for other scripts.

Seems this may need some reworking to account for this and make sure it only captures options specified after the script name. You might also need to take into account the fact that the script name itself may contain `-i` or similar flags within the script path itself if you do decide to go for a regex approach to this.

Reply:
Ah you are right. Apologies for not realising that earlier. 👍   I'll try to come up with a fix for this. Thanks so much for being so thorough in this.

Reply:
This should be fixed after my latest force-push. I correctly get the index of the script name in the `args` array, and get the arguments after the script name only. 👍